File: seungmi/week10/징검다리.py
def solution(distance, rocks, n):
    answer = 0
    start, end = 0, distance
    
    rocks.sort()
    
    #이분 탐색
    while start <= end: 
        mid = (start + end) // 2
        del_stones = 0 #제거한 돌을 카운트하기 위한 변수
        pre_stone = 0 #기준이되는 돌(시작지점)
        for rock in rocks:
            if rock - pre_stone <  mid: 
                del_stones += 1 
            else:
                pre_stone = rock
            if del_stones > n:
            	break
                
        if del_stones > n: 
            end = mid - 1
        else:
            answer = mid
            start = mid + 1
            
    return answer

File: sujeong/week10/징검다리.swift
import Foundation

func getNumOfRemovedStone(_ minGap:Int, _ rocks:[Int]) -> Int {
    var here:Int = 0
    var removeCnt:Int = 0
    for rock in rocks {
        if rock - here < minGap {
            removeCnt += 1
        } else {
            here = rock
        }
    }
    return removeCnt
}

func solution(_ distance:Int, _ rocks:[Int], _ n:Int) -> Int {
    var answer:Int = 0
    let sortedRocks:[Int] = rocks.sorted()+[distance]
    
    var start:Int = 0
    var end:Int = distance
    var mid:Int = (start + end)/2
    
    while start <= end {
        mid = (start+end)/2
        
        let removeCnt:Int = getNumOfRemovedStone(mid, sortedRocks)
        if removeCnt > n {
            end = mid-1
        } else {
            start = mid+1
        }
    }
    return end
}
